### ISSUE TITLE
IIASA format write-out crashing

### ISSUE THREAD

**Issue by pkyle** [JGCRI/gcamrpt#55]
I'm working on the transportation branch but the code looks the same on master, so this may be a problem for others too. If I try to write out the data in IIASA format (i.e., dataformat == 'IIASA'), I get the following crash after all of the queries have been run and all data aggregated:
```
Error in if (nrow(vardf) == 0) rslts[var] <- NULL : 
  argument is of length zero
```
The problem stems from line 221 in mcl.R, where the following line:
```
vardf <- rslts[var]
```
doesn't make a data frame; it makes a list of length one whose first element is a data frame. So, the next line, which is intended to check the number of rows of the data frame, instead causes a crash.

**Comment by kmulvaney** [comment]
Hi, @pkyle.  I am working on the "air pollution" branch, and I am able to specify the IIASA dataformat without crashing.  The air pollution branch is up to date here on Github.  If you have any questions, let me know.

**Comment by pkyle** [comment]
Interesting! I see the same code on the air_pollution branch (albeit in line 242). @kmulvaney are you merging scenarios (i.e., scenmerge = TRUE)? If not then I won't propose a fix; this might just reflect different conventions in different versions of R or operating systems or whatever. The code that's causing mine to crash is inside an `if(scenmerge)` block. In mine, `if(nrow(vardf) == 0)` is effectively the same as `if(NULL == 0)`, which resolves to `if(logical(0))`. So, for an example of how R responds to that:
```
> if(logical(0)) print("ok")
Error in if (logical(0)) print("ok") : argument is of length zero
```

**Comment by kmulvaney** [comment]
@pkyle - I do not merge scenarios when I am using gcamrpt for my research.  When scenmerge = FALSE, the IIASA data format option works fine for me.  However, when I set scenmerge = TRUE, I get the same error that you describe in your first comment.

**Comment by rplzzz** [comment]
I believe this was fixed in #56 .
